ofdb-entities/src/email.rs

```diff
@@ -1,62 +1,85 @@
 use std::{borrow::Borrow, fmt, ops::Deref, str::FromStr};
+use thiserror::Error;
 
-// TODO: rename to EmailAddress
 #[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
-pub struct Email(String);
+pub struct EmailAddress(String);
 
-impl AsRef<String> for Email {
+impl EmailAddress {
+    pub const fn from_string_unchecked(address: String) -> Self {
+        Self(address)
+    }
+}
+
+impl AsRef<String> for EmailAddress {
     fn as_ref(&self) -> &String {
         &self.0
     }
 }
 
-impl AsRef<str> for Email {
+impl AsRef<str> for EmailAddress {
     fn as_ref(&self) -> &str {
         self.0.as_str()
     }
 }
 
-impl From<Email> for String {
-    fn from(from: Email) -> Self {
+impl From<EmailAddress> for String {
+    fn from(from: EmailAddress) -> Self {
         from.0
     }
 }
 
-impl From<String> for Email {
+// FIXME: Remove
+impl From<String> for EmailAddress {
     fn from(from: String) -> Self {
         Self(from)
     }
 }
 
-impl From<&str> for Email {
+// FIXME: Remove
+impl From<&str> for EmailAddress {
     fn from(from: &str) -> Self {
         from.to_owned().into()
     }
 }
 
-impl FromStr for Email {
-    type Err = ();
-    fn from_str(s: &str) -> Result<Email, Self::Err> {
-        Ok(s.into())
+#[derive(Debug, Error)]
+#[error("Invalid E-Mail address")]
+pub struct EmailParseError;
+
+impl FromStr for EmailAddress {
+    type Err = EmailParseError;
+    fn from_str(s: &str) -> Result<EmailAddress, Self::Err> {
+        let email_address = mailparse::addrparse(s)
+            .ok()
+            .and_then(|list| list.extract_single_info())
+            .ok_or(EmailParseError)?
+            .addr;
+        Ok(Self(email_address))
     }
 }
 
-impl Borrow<str> for Email {
+impl Borrow<str> for EmailAddress {
     fn borrow(&self) -> &str {
         self.as_ref()
     }
 }
 
-impl Deref for Email {
+impl Deref for EmailAddress {
```

Comment:
Do we really want to implement `Deref`? Using string operations on e-mail addresses should not occur very often and then you better borrow the inner string  explicitly.